README.md

```diff
@@ -53,98 +53,102 @@ If you need to install Go, follow [the official instructions](https://golang.org
-    VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
-      AddressSpace: &network.AddressSpace{
-        AddressPrefixes: &[]string{"10.0.0.0/8"},
-      },
-      Subnets: &[]network.Subnet{
-        {
-          Name: to.StringPtr("<subnet1Name>"),
-          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
-            AddressPrefix: to.StringPtr("10.0.0.0/16"),
-          },
-        },
-        {
-          Name: to.StringPtr("<subnet2Name>"),
-          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
-            AddressPrefix: to.StringPtr("10.1.0.0/16"),
-          },
-        },
-      },
-    },
-  },
-  nil)
+func main() {
+	vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+
+	if err == nil {
+		vnetClient.Authorizer = authorizer
+	}
+
+	vnetClient.CreateOrUpdate(context.Background(),
+		"<resourceGroupName>",
+		"<vnetName>",
+		network.VirtualNetwork{
+			Location: to.StringPtr("<azureRegion>"),
+			VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+				AddressSpace: &network.AddressSpace{
+					AddressPrefixes: &[]string{"10.0.0.0/8"},
+				},
+				Subnets: &[]network.Subnet{
+					{
+						Name: to.StringPtr("<subnet1Name>"),
+						SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+							AddressPrefix: to.StringPtr("10.0.0.0/16"),
+						},
+					},
+					{
+						Name: to.StringPtr("<subnet2Name>"),
+						SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+							AddressPrefix: to.StringPtr("10.1.0.0/16"),
+						},
+					},
+				},
+			},
+		})
+}
 ```
 
 ### Authentication
 
-Most SDK operations require an OAuth token for authentication and authorization.
+Most SDK operations require an OAuth token for authentication and authorization. These are
+made available in the Go SDK For Azure through types implementing the `Authorizer` interface.
 You can get one from Azure Active Directory using the SDK's
-[authentication](https://godoc.org/github.com/Azure/go-autorest/autorest/adal) package
-and an existing service principal (aka Client ID and secret) as shown in the
-following example.
-
-The `OAuthTokenProvider` returned by this sample should be set as the
-authorizer for the resource client, as shown in the [previous section](#use).
-
-Note: To create a new service principal, run
+[authentication](https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth) package. The `Authorizer` returned  should
+be set as the authorizer for the resource client, as shown in the [previous section](#use).
+
+You can get an authorizer in the following ways:
+1. From the **Environment**:
+  - Use `auth.auth.NewAuthorizerFromEnvironment()`. This call will try to get an authorizer based on the environment
+variables in the following order: Client Credentials, Client Certificate, Username Pasword, MSI.
+  - The environment variables available are:
```

Comment:
>The environment variables available are: [](start = 4, length = 40)

Maybe it is a good idea to group env vars depending on which kind of auth they are used. This way it is more organized and less text

Reply:
Yea I was thinking of doing that but there are a couple of shared env variables and it felt like duplicating a little.

Reply:
Ah, you can write that x, y, z vars are needed in all cases

Reply:
well not all cases, just shared :)